src/pages/index.js

```diff
@@ -0,0 +1,266 @@
+const cardModalNameInput = cardModal.querySelector(`#card-link-input-caption`);
+
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImage = previewModal.querySelector(".modal__image");
+const previewModalPicDescription = previewModal.querySelector(
+  ".modal__pic-description"
+);
+const previewModalCloseBtn = previewModal.querySelector(
+  ".modal__close-btn_type_preview"
+);
+
+const cardTemplate = document.querySelector(`#card-template`);
+const cardsList = document.querySelector(`.cards__list`);
+
+let selectedCard, selectedCardId;
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "3e36594c-9f58-43ca-8212-d4a4b67e7c65",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    cards.forEach(function (ArrayItem) {
+      const cardElement = getCardElement(ArrayItem);
+      cardsList.prepend(cardElement);
+    });
+    profileName.textContent = userInfo.name;
+    profileAvatar.alt = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+  })
+  .catch(console.error);
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(`.card`)
+    .cloneNode(true);
+  const cardDeleteBtn = cardElement.querySelector(`.card__delete-btn`);
+  const cardNameElement = cardElement.querySelector(`.card__title`);
+  const cardImageElement = cardElement.querySelector(`.card__image`);
+  const cardLikeBtn = cardElement.querySelector(`.card__like-button`);
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-button_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => {
+    handleCardLike(evt, data._id);
+  });
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageElement.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImage.src = data.link;
+    previewModalImage.alt = data.name;
+    previewModalPicDescription.textContent = data.name;
+  });
+  return cardElement;
+}
+function handleCardLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalEscape);
+}
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalEscape);
+}
+previewModalCloseBtn.addEventListener("click", () => {
```

Comment:
**Could be improved**


You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.